drivers/pca9633/pca9633.c

```diff
@@ -0,0 +1,266 @@
+ * @param[in] data      Data to write
+ *
+ * @return  0 on success
+ * @return  -PCA9633_ERROR_I2C if acquiring of I2C bus fails
+ * @return  -EIO When slave device doesn't ACK the byte
+ * @return  -ENXIO When no devices respond on the address sent on the bus
+ * @return  -ETIMEDOUT When timeout occurs before device's response
+ * @return  -EINVAL When an invalid argument is given
+ * @return  -EOPNOTSUPP When MCU driver doesn't support the flag operation
+ * @return  -EAGAIN When a lost bus arbitration occurs
+ */
+int _write_reg(pca9633_t* dev, uint8_t reg, uint8_t data);
+
+/**
+ * @brief Read data from a register.
+ *
+ * @param[in] dev       Device descriptor of the PCA9633
+ * @param[in] reg       Register address to read from
+ * @param[out] data     Byte read from given registerAddress
+ *
+ * @return  0 on success
+ * @return  -PCA9633_ERROR_I2C if acquiring of I2C bus fails
+ * @return  -EIO When slave device doesn't ACK the byte
+ * @return  -ENXIO When no devices respond on the address sent on the bus
+ * @return  -ETIMEDOUT When timeout occurs before device's response
+ * @return  -EINVAL When an invalid argument is given
+ * @return  -EOPNOTSUPP When MCU driver doesn't support the flag operation
+ * @return  -EAGAIN When a lost bus arbitration occurs
+ */
+int _read_reg(pca9633_t* dev, uint8_t reg, uint8_t* data);
+
+int pca9633_init(pca9633_t *dev, pca9633_params_t *params)
+{
+    assert(dev);
+    assert(params);
+
+    dev->params = *params;
+
+    i2c_init(dev->params.i2c_dev);
+
+    int rc = _write_reg(dev, PCA9633_REG_MODE1, 0x0);
+    _write_reg(dev, PCA9633_REG_MODE2, 0x0);
+
+    if (rc != PCA9633_OK) {
+        return rc;
+    }
+
+    pca9633_set_ldr_state_all(dev, PCA9633_LDR_STATE_IND_GRP);
+    pca9633_set_group_control_mode(dev, PCA9633_GROUP_CONTROL_MODE_DIMMING);
+    pca9633_set_rgb(dev, 255, 255, 255);
+
+    return PCA9633_OK;
+}
+
+void pca9633_wakeup(pca9633_t* dev)
+{
+    uint8_t prev_reg;
+    _read_reg(dev, PCA9633_REG_MODE1, &prev_reg);
+    uint8_t newReg = prev_reg & ~(1 << PCA9633_BIT_SLEEP);
+
+    _write_reg(dev, PCA9633_REG_MODE1, newReg);
+}
+
+void pca9633_sleep(pca9633_t* dev)
+{
+    uint8_t prev_reg;
+    _read_reg(dev, PCA9633_REG_MODE1, &prev_reg);
+    uint8_t newReg = prev_reg | (1 << PCA9633_BIT_SLEEP);
+
+    _write_reg(dev, PCA9633_REG_MODE1, newReg);
+}
+
+void pca9633_turn_on(pca9633_t* dev)
+{
+    _write_reg(dev, PCA9633_REG_LEDOUT, dev->stored_reg_ledout);
+}
+
+void pca9633_turn_off(pca9633_t* dev)
+{
+    _read_reg(dev, PCA9633_REG_LEDOUT, &dev->stored_reg_ledout);
+    _write_reg(dev, PCA9633_REG_LEDOUT, PCA9633_LDR_STATE_OFF);
+}
+
+void pca9633_set_pwm(pca9633_t* dev, uint8_t reg_pwm, uint8_t pwm)
+{
+    _write_reg(dev, reg_pwm, pwm);
+}
+
+void pca9633_set_grp_pwm(pca9633_t* dev, uint8_t pwm)
+{
+    _write_reg(dev, PCA9633_REG_GRPPWM, pwm);
+}
+
+void pca9633_set_blinking(pca9633_t* dev, uint8_t blink_period,
+        float on_off_ratio)
```

Comment:
```suggestion
void pca9633_set_blinking(pca9633_t* dev, uint8_t blink_period,
        uint8_t on_off_ratio)
```

`float` is only available on Cortex-M4F and above - no need to pull in soft float libraries when you just want to use a value between 0 and 255.

Reply:
I changed it to an uint8_t